servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -1117,11 +1250,89 @@ public void reportLostFile(long fileId) throws FileDoesNotExistException {
   }
 
   public List<Integer> getPriorityDependencyList() {
-    synchronized (mDependencyMap) {
-      return mDependencyMap.getPriorityDependencyList();
+    return mDependencyMap.getPriorityDependencyList();
+  }
+
+  public long loadFileInfoFromUfs(TachyonURI path, boolean recursive)
+      throws BlockInfoException, FileAlreadyExistException, FileDoesNotExistException,
+      InvalidPathException, SuspectedFileSizeException, TachyonException {
+    TachyonURI ufsPath;
+    synchronized (mInodeTree) {
+      ufsPath = mMountTable.resolve(path);
+    }
+    UnderFileSystem ufs = UnderFileSystem.get(ufsPath.toString(), MasterContext.getConf());
+    try {
+      long ufsBlockSizeByte = ufs.getBlockSizeByte(ufsPath.toString());
+      long fileSizeByte = ufs.getFileSize(ufsPath.toString());
+      // Metadata loaded from UFS has no TTL set.
+      long fileId = createFile(path, ufsBlockSizeByte, recursive, Constants.NO_TTL);
+      persistFile(fileId, fileSizeByte);
+      return fileId;
+    } catch (IOException e) {
+      LOG.error(ExceptionUtils.getStackTrace(e));
+      throw new TachyonException(e.getMessage());
     }
   }
 
+  public boolean mount(TachyonURI tachyonPath, TachyonURI ufsPath)
+      throws FileAlreadyExistException, InvalidPathException, IOException {
+    synchronized (mInodeTree) {
+      mkdirs(tachyonPath, true);
+      if (mountInternal(tachyonPath, ufsPath)) {
+        writeJournalEntry(new AddMountPointEntry(tachyonPath, ufsPath));
+        flushJournal();
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void mountFromEntry(AddMountPointEntry entry) {
+    TachyonURI tachyonPath = entry.getTachyonPath();
+    TachyonURI ufsPath = entry.getUfsPath();
+    try {
+      if (!mountInternal(tachyonPath, ufsPath)) {
+        LOG.error("Failed to mount " + ufsPath + " at " + tachyonPath);
+      }
+    } catch (InvalidPathException e) {
+      LOG.error("Failed to mount " + ufsPath + " at " + tachyonPath);
```

Comment:
I was, but I guess to be consistent with the behavior of other ``...FromEntry`` methods, we should just throw it. Changed.